### ISSUE TITLE
AKS ClusterRoleBinding is not created for system:azure-cloud-provider

### ISSUE THREAD

**Issue by nopjmp** [Azure/AKS#1104]
**What happened**:

1. Tried to delete a load balancer
2. Load balancer failed to delete since it was attached to a public IP
3. Event wasn't logged as the azure-cloud-provider service account didn't have the right permissions.

`I0712 12:37:07.439245 1 authorization.go:73] Forbidden: "/api/v1/namespaces/[redacted]/events/[redacted]", Reason: "no RBAC policy matched"`

**What you expected to happen**:

Failure events to show up in `kubectl get events -A`

**How to reproduce it (as minimally and precisely as possible)**:

1. Create a new cluster
2. Check for ClusterRoleBinding system:azure-cloud-provider

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`): Server Version: version.Info{Major:"1", Minor:"12", GitVersion:"v1.12.8", GitCommit:"a89f8c11a5f4f132503edbc4918c98518fd504e3", GitTreeState:"clean", BuildDate:"2019-04-23T04:41:47Z", GoVersion:"go1.10.8", Compiler:"gc", Platform:"linux/amd64"}
- Size of cluster (how many worker nodes are in the cluster?): 3
- HTTP Microservices

**Workaround**:

`kubectl apply -f https://raw.githubusercontent.com/Azure/aks-engine/master/parts/k8s/addons/kubernetesmasteraddons-azure-cloud-provider-deployment.yaml`

**Comment by jnoller** [comment]
Is this an aks-engine or AKS cluster? We have not seen any other reports of issues like this

**Comment by nopjmp** [comment]
This is an AKS cluster. It's not a major issue as the effect that happens is the events are not properly logged. We didn't realize what the error was until I resolved this permissions problem. It's in the kubenetes repo for the plugin as well as aks-engine to be applied to the cluster, but it doesn't seem to be applied when the cluster is created.

Without this permissions, `kubectl get events` won't return back the error messages to find them easily when the provider fails to delete a load balancer for example. This is how I stumbled upon this in Azure.
